src/core/mac/sub_mac.cpp

```diff
@@ -487,7 +495,12 @@ void SubMac::HandleTransmitDone(TxFrame &aFrame, RxFrame *aAckFrame, Error aErro
         {
             mCallbacks.RecordCcaStatus(ccaSuccess, aFrame.GetChannel());
         }
-
+#if OPENTHREAD_CONFIG_MAC_CSL_RECEIVER_ENABLE
+        if (mState == kStateCslSample)
+        {
+            mCslLastSync = TimerMicro::GetNow();
```

Comment:
Invalid state check. CSL receiver does not need to wait for `kStateCslSample` state to send packet to its parent. The router keeps its radio on and SSED can send data packet to it whenever it wants.